astropy/units/structured.py

```diff
@@ -0,0 +1,410 @@
+                                                         other.values())]
+
+        def converter(value):
+            if not hasattr(value, 'dtype'):
+                value = np.array(value, self._recursively_get_dtype(value))
+            result = np.empty_like(value)
+            for name, converter_ in zip(result.dtype.names, converters):
+                result[name] = converter_(value[name])
+            return result
+
+        return converter
+
+    def to(self, other, value, equivalencies=[]):
+        """Return values converted to the specified unit.
+
+        Parameters
+        ----------
+        other : `~astropy.units.StructuredUnit`
+            The unit to convert to.  If necessary, will be converted to
+            a Structured Unit using the dtype of ``value``.
+        value : array-like
+            Value(s) in the current unit to be converted to the
+            specified unit.  If a sequence, the first element must have
+            entries of the correct type to represent all elements (i.e.,
+            not have, e.g., a ``float`` where other elements have ``complex``).
+        equivalencies : list of tuple, optional
+            A list of equivalence pairs to try if the units are not
+            directly convertible.  See :ref:`unit_equivalencies`.
+            This list is in addition to possible global defaults set by, e.g.,
+            `set_enabled_equivalencies`.
+            Use `None` to turn off all equivalencies.
+
+        Returns
+        -------
+        values : scalar or array
+            Converted value(s).
+
+        Raises
+        ------
+        UnitsError
+            If units are inconsistent
+        """
+        return self._get_converter(other, equivalencies=equivalencies)(value)
+
+    def to_string(self, format='generic'):
+        """Output the unit in the given format as a string.
+
+        Units are separated by commas.
+
+        Parameters
+        ----------
+        format : `astropy.units.format.Base` instance or str
+            The name of a format or a formatter object.  If not
+            provided, defaults to the generic format.
+        """
+        if format not in ('generic', 'unscaled', 'latex'):
+            raise ValueError("Structured units cannot be written in {0} "
+                             "format. Only 'generic', 'unscaled' and "
+                             "'latex' are supported.".format(format))
+        out = '({})'.format(', '.join([part.to_string(format)
+                                       for part in self.values()]))
+        return out if len(self) > 1 else out[:-1] + ',)'
+
+    def _repr_latex_(self):
+        return self.to_string('latex')
+
+    __array_ufunc__ = None
+
+    def __mul__(self, other):
+        if isinstance(other, str):
+            try:
+                other = Unit(other, parse_strict='silent')
+            except Exception:
+                return NotImplemented
+        if isinstance(other, UnitBase):
```

Comment:
FunctionUnitBase?

Reply:
I hadn't thought about that, but as one cannot usefully multiply with a logarithmic unit, this one happens to be OK...

Reply:
SGTM